### PR TITLE
Hmpid pedestals v3

### PR DESCRIPTION
**Pull request by fapfap69**

### CODE REVIEW

**Comment by martenole** [comment]
Hi @fapfap69 seems there are some conflicts, could you check please? And did you already try to run with your local O2DPG to see if the calibration works now?

**Comment by fapfap69** [comment]
Hi Ole,
the conflict is related to the deletion of extra environment variables in addW() macro.

<<<<<<< HMPID-PedestalsV3
add_W o2-dpl-raw-proxy "--dataspec \"$PROXY_INSPEC\" --channel-config \"name=readout-proxy,type=pull,method=connect,address=ipc://@$INRAWCHANNAME,rateLogging=1,transport=shmem\""
add_W o2-hmpid-raw-to-pedestals-workflow "--fast-decode $SPEC_PARAM"
=======
add_W o2-dpl-raw-proxy "$ARGS_ALL --dataspec \"$PROXY_INSPEC\" --channel-config \"name=readout-proxy,type=pull,method=connect,address=ipc://@$INRAWCHANNAME,rateLogging=1,transport=shmem\""
add_W o2-hmpid-raw-to-pedestals-workflow "${ARGS_ALL} --configKeyValues \"$ARGS_ALL_CONFIG\" --fast-decode $SPEC_PARAM"
>>>>>>> master

... I could try to delete the pull request, rebase my branch and redo the PR.

Nevertheless yesterday I try to run the Pedestal (with an updated version on my personal repo), but an other error appeared.
Unfortunately, I loose the error message.

Today, as soon as possible, I'm going to redo the Pedestal-run, in order to catch the error message and fix it (if I shall able to do it :(


Antonio

**Comment by martenole** [comment]
Hi Antonio,
ok, I would try to make it work with your personal repo first and then update this PR (or close it and open a new one if you prefer). Let me know in case there is something I can help you with.
Cheers,
Ole

**Comment by fapfap69** [comment]
Hi Ole,

I re-tray to run just now the : ../calib/hmp-pedestals-processing.sh

the error is related to the --configKeyValues option on the o2-dpl-raw-proxy, that the parser doesn't recognise

but, actually in my repo the slice of code is 
---------------------------------
WORKFLOW=
add_W o2-dpl-raw-proxy "--dataspec \"$PROXY_INSPEC\" --channel-config \"name=readout-proxy,type=pull,method=connect,address=>
add_W o2-hmpid-raw-to-pedestals-workflow "--fast-decode $SPEC_PARAM"
---------------------------------

in which that option was deleted.


Seems that the executed script is the old one, and that it isn't read from my repo.

The ECS configuration is

**Comment by fapfap69** [comment]
Hi Ole,
this morning I was able to run the Pedestals-run, but the error stay there:
==========
2023-09-11 13:20:56.105013 inf  odc-grpc-server     472760 2hbxqNw7ujX:0        GEN_TOPO_SOURCE='/home/fap/O2DPG/DATA'
2023-09-11 13:20:56.105022 inf  odc-grpc-server     472760 2hbxqNw7ujX:0        DDMODE='processing'
2023-09-11 13:20:56.105030 inf  odc-grpc-server     472760 2hbxqNw7ujX:0        GEN_TOPO_LIBRARY_FILE='production/standalone-calibration.desc'
.
.
.
2023-09-11 13:20:59.025988 fat  odc-grpc-server     472760 2hbxqNw7ujX:0        Found error messages during topology generation
2023-09-11 13:20:59.025997 fat  odc-grpc-server     472760 2hbxqNw7ujX:0        Non-zero return value 1 during topology generation
2023-09-11 13:20:59.026006 fat  odc-grpc-server     472760 2hbxqNw7ujX:0        Error running command GEN_TOPO_CALIB_WORKFLOW=0 SHMSIZE=180388626432 EPN_NODE_MI100=1 EPN_GLOBAL_SCALING=\"3/2\" WORKFLOWMODE_FILE=/tmp/o2_workflow_1oyf1gyr/wf1.dds GLOBALDPLOPT+=\" -b --dds-workflow-suffix _reco1\"  production/calib/hmp-pedestals-processing.sh
2023-09-11 13:20:59.026018 fat  odc-grpc-server     472760 2hbxqNw7ujX:0        stderr:
2023-09-11 13:20:59.026027 fat  odc-grpc-server     472760 2hbxqNw7ujX:0        stdout: [ERROR] error parsing options of o2-dpl-raw-proxy: unrecognised option '--configKeyValues'
2023-09-11 13:20:59.026035 fat  odc-grpc-server     472760 2hbxqNw7ujX:0
2023-09-11 13:20:59.026044 fat  odc-grpc-server     472760 2hbxqNw7ujX:0        Execution time:  0.3930337429046631
2023-09-11 13:20:59.026052 fat  odc-grpc-server     472760 2hbxqNw7ujX:0        Found error messages during topology generation
========

I did the rebase of my repo (fap) and also put the modifications in the master branch.
The parser find '--configKeyValues' option that is not in the code :(

========
# Start with an empty workflow
WORKFLOW=
add_W o2-dpl-raw-proxy "--dataspec \"$PROXY_INSPEC\" --channel-config \"name=readout-proxy,type=pull,method=connect,address=ipc://@$INRAWCHANNAME,rateLogging=1,transport=shmem\""
add_W o2-hmpid-raw-to-pedestals-workflow "--fast-decode $SPEC_PARAM"

# Finally add the o2-dpl-run workflow manually, allow for either printing the workflow or creating a topology (default)
WORKFLOW+="o2-dpl-run ${ARGS_ALL} ${GLOBALDPLOPT}"
========

I'm not understand where is the 'sh' file that the parser analysed.

Cheers

Antonio

File: DATA/production/calib/hmp-pedestals-processing.sh [hunk 1]
```diff
@@ -53,35 +52,35 @@ then
...
 
-WORKFLOW+="--files-basepath 'HMP' "
-WORKFLOW+="--pedestals-tag ${HMP_PED_TAG} --sigmacut ${HMP_SIGMACUT} | "
+SPEC_PARAM+="--files-basepath 'HMP' "
+SPEC_PARAM+="--pedestals-tag ${HMP_PED_TAG} --sigmacut ${HMP_SIGMACUT}"
 
+#  Here we compose the workflow
+# Start with an empty workflow
+WORKFLOW=
+add_W o2-dpl-raw-proxy "--dataspec \"$PROXY_INSPEC\" --channel-config \"name=readout-proxy,type=pull,method=connect,address=ipc://@$INRAWCHANNAME,rateLogging=1,transport=shmem\""
```
Comment (martenole) [inline DATA/production/calib/hmp-pedestals-processing.sh:76]:
```suggestion
add_W o2-dpl-raw-proxy "--dataspec \"$PROXY_INSPEC\" --channel-config \"name=readout-proxy,type=pull,method=connect,address=ipc://@$INRAWCHANNAME,rateLogging=1,transport=shmem\"" "" 0
```
Hi @fapfap69 I have tried the topology generation with the above change and it works. The problem is that by defaul the `add_W` function is adding the `ARGS_ALL_CONFIG ` to the workflow. To disable it one has to add the zero at the end: https://github.com/AliceO2Group/O2DPG/blob/4750f15e3eaca59ac3325d0580fdf8e2c7cb8a99/DATA/common/gen_topo_helper_functions.sh#L223
Sorry I did not spot this earlier

**Comment by fapfap69** [comment]
Solved, it works fine !

Thanks a lot 

Antonio

**Comment by martenole** [comment]
Very good :)
Could you then please rebase your branch to fix the merge conflicts? Then we can merge your PR

**Pull request merged**